snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -158,11 +161,26 @@ def get_build_commands(self) -> List[str]:
 
         commands.extend(["echo 'Built Matter SDK'"])
 
-        # Compare and output pigweed related environment variables to matter_sdk_env env file
+        # Compare the difference between the original PATH and the modified PATH
+        commands.extend(
+            [
+                'IFS=: read -r -a OLD_PATH_ARRAY <<< "$OLD_PATH"',
+                'IFS=: read -r -a NEW_PATH_ARRAY <<< "$PATH"',
+                "declare -a DIFFERENCE=()",
+                'for element in "${NEW_PATH_ARRAY[@]}"; do',
+                '    if [[ ! " ${OLD_PATH_ARRAY[@]} " =~ " $element " ]]; then',
+                '        DIFFERENCE+=("$element")',
+                "    fi",
+                "done",
+                'MATTER_SDK_PATH=$(IFS=:; echo "${DIFFERENCE[*]}")',
```

Comment:
I find this very convoluted. How about using [shell parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html):
```bash
MATTER_SDK_PATHS="${PATH%$OLD_PATH}"
```
and then `'echo "export PATH=$MATTER_SDK_PATHS\\$PATH" >> matter-sdk-env.sh',` ?

Note that `MATTER_SDK_PATHS` will include the separator.